assets/js/custom/Translation.js

```diff
@@ -2,6 +2,137 @@
 
 // eslint-disable-next-line no-unused-vars
 function Translation (hasDescription, hasCredit) {
+  const providerMap = {
+    'itranslate': 'iTranslate'
+  }
+  let languageMap = {}
+  let targetLanguage = document.documentElement.lang
+  let creditText = 'Translated by %provider% from %sourceLanguage% to %targetLanguage%'
+
+  $(document).ready(function () {
+    setTargetLanguage()
+    getLanguageMap()
+    getTranslationCreditText()
+  })
+
+  $(document).on('click', '.comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
+
+    $(this).hide()
+
+    if (isTranslationAvailable(commentId)) {
+      $('#comment-translation-loading-spinner-' + commentId).show()
+      translateComment(commentId)
+    } else {
+      openTranslatedComment(commentId)
+    }
+  })
+
+  $(document).on('click', '.remove-comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('remove-comment-translation-button-'.length)
+    $(this).hide()
+    $('#comment-translation-button-' + commentId).show()
+    $('#comment-translation-wrapper-' + commentId).slideUp()
+    $('#comment-text-wrapper-' + commentId).slideDown()
+  })
+
+  function setTargetLanguage () {
+    const decodedCookie = decodeURIComponent(document.cookie).split(';')
+    for (let i = 0; i < decodedCookie.length; i++) {
+      if (decodedCookie[i].includes("hl=") && decodedCookie[i].length < 10) {
+        targetLanguage = decodedCookie[i].substring(decodedCookie[i].indexOf("=") + 1).replace('_', '-')
+      }
+    }
+    console.log(targetLanguage)

```

Comment:
Looks like there are still some refactoring left in this function

Reply:
Removed log